src/main/java/com/wolox/training/services/BookService.java

```diff
@@ -102,4 +132,17 @@ private void adaptBookDtoToBookModel(BookDTO bookDto, Book book) {
         book.setTitle(bookDto.getTitle());
         book.setYear(bookDto.getYear());
     }
+
+    private void adaptBookInfoToBookModel(BookInfoDTO bookInfo, Book book) {
+        book.setIsbn(bookInfo.getIsbn());
+        book.setYear(bookInfo.getPublishDate());
+        book.setAuthor(bookInfo.getAuthors().toString());
+        book.setPublisher(bookInfo.getPublishers().toString());
+        book.setTitle(bookInfo.getTitle());
+        book.setSubtitle(bookInfo.getSubtitle());
+        book.setPages(bookInfo.getNumberOfPages());
+        book.setImage("-");
```

Comment:
Do you need to set this? Aren't you doing it directly in the User class as initialization?